bin/yaml/cpp.yaml

```diff
@@ -722,6 +722,46 @@ compilers:
           - 14.0.0
           - 15.0.0
           - 16.0.0
+    clang-assert:
+      - type: tarballs
+        check_exe: bin/clang --version
+        dir: clang-{name}-assert
+        url: https://storage.yandexcloud.net/clang-assert/clang-{name}-assert.tar.xz
+        compression: xz
+        targets:
+          - name: 2.6.0
+      - type: tarballs
+        check_exe: bin/clang++ --version
+        dir: clang-{name}-assert
+        url: https://storage.yandexcloud.net/clang-assert/clang-{name}-assert.tar.xz
+        compression: xz
+        targets:
```

Comment:
I'm a little wary of relying on _all_ clang compilers coming from an external source. We take great care in knowing the provenance of all our main compilers and this would break that for arguably the joint-most important.

I appreciate I was rather distracted during the discussion of this: is there no way for the at least the more contemporary compilers to be built using our own infrastructure?